sage_modabvar/lseries.py

```diff
@@ -104,20 +104,26 @@ def __call__(self, s):
             sage: L(1)
             0.386769938387780
         """
-        abelian_variety = self.abelian_variety()
-        # Check for easy J0 case
-        if is_Gamma0(abelian_variety.group()) and abelian_variety.is_ambient():
-            S = CuspForms(abelian_variety.level())
-            newforms = S.newforms('a')
-        else:
-            simples = abelian_variety.decomposition()
-            newforms = [Newform(simple.newform_label(),'a')
-                    for simple in simples]
-
-        factors = [newform.lseries(embedding=i)
-                for newform in newforms
-                for i in range(newform.base_ring().degree())]
-        return prod(L(s) for L in factors)
+        try:
+            factors = self.__factors
+            return prod(L(s) for L in factors)
+        except AttributeError:
+            abelian_variety = self.abelian_variety()
+            # Check for easy J0 case
+            if is_Gamma0(abelian_variety.group()) \
+            and abelian_variety.is_ambient():
+                S = CuspForms(abelian_variety.level())
+                newforms = S.newforms('a')
+            else:
+                simples = abelian_variety.decomposition()
+                newforms = [Newform(simple.newform_label(), 'a')
+                        for simple in simples]
+
+            factors = [newform.lseries(embedding=i)
+                    for newform in newforms
+                    for i in range(newform.base_ring().degree())]
+            self.__factors = factors
+            return prod(L(s) for L in factors)
```

Comment:
unindent this line.  

Then you have eliminated one line of code that was repeated twice. If you ever change it or find a bug, it's less likely that you'll have to fix the other copy of that line.